### ISSUE TITLE
ServiceWorker needs to allow some requests to go through to e.g. a backend

### ISSUE THREAD

**Issue by jakhog** [dolittle-interaction/WebAssembly#44]
In the current setup, the ServiceWorker will intercept all requests that go through the same domain as the web-app itself. Which means that any requests to a cloud-based backend (e.g. for syncing data when the cloud is reachable), it will be stopped by the ServiceWorker because it doesn't allow any POSTs.

We need a mechanism to allow some URLs to pass straight through the ServiceWorker without any interruptions.

**Comment by einari** [comment]
We have the WebPack generator plugin that deals with this. We could let the plugin have options that could then pick this up.
Would this work?

```javascript
config.plugins.push(new ServiceWorkerGenerator({
    excludedRelativeUrls : [
          "/some/fragment",
          "/some/other/fragment"
    ]
}));
```

It would then add an array in the service worker of those URL fragments it would skip for caching.​

**Comment by jakhog** [comment]
That sound like it would work perfectly! As long as it doesn't care which HTTP method is used.

How would we configure this? Manually in the webpack.config?

**Comment by smithmx** [comment]
Can't we just add another url?  So if the app is waterproof.wihelmsen.com then it uploads to cloud-twin.waterproof.wilhelmsen.com.  Or am I misunderstanding the issue?

**Comment by jakhog** [comment]
We could do that, and we have done that 🙂 So this is not a must-fix-now issue.

But, with this approach we end up doing Cross-Origin-Requests, so it means we have to allow and deal with that.

Personally, I also like it when the URLs have the application as the hostname, and split bounded contexts with the path.

**Comment by einari** [comment]
Let's try to avoid cross origin if we can. Since we can route on paths, it makes sense to do that.

I thought this was just about letting a specific path through.
We can get back to this when we generalize the sync of events - which is the use case we discuss right now, right?

**Comment by jakhog** [comment]
Yes, this is just about letting a specific path through, on the same host/origin as a ServiceWorker is registered for an offline app. I think that's a capability we want to support for other reasons other than syncing events as well (although we are definitely going to need it for that).
